File: pollfish-admob-adapter/example/src/main/java/com/pollfish/example/MainActivity.java
package com.pollfish.example;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.gms.ads.AdError;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.FullScreenContentCallback;
import com.google.android.gms.ads.LoadAdError;
import com.google.android.gms.ads.MobileAds;
import com.google.android.gms.ads.rewarded.RewardedAd;
import com.google.android.gms.ads.rewarded.RewardedAdLoadCallback;
import com.pollfish.mediation.PollfishAdMobAdapter;
import com.pollfish.mediation.PollfishExtrasBundleBuilder;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";

    //TODO: Replace with your own keys
    private static final String AD_MOB_KEY = "ADMOB_AD_UNIT_KEY";
    private static final String POLLFISH_API_KEY = "POLLFISH_API_KEY";

    private RewardedAd mRewardedAd;
    private Button mRewardedAdButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);

        mRewardedAdButton = findViewById(R.id.rewardedAdBtn);

        MobileAds.initialize(this, initializationStatus -> {
            Log.d(TAG, "onInitializationComplete()");
            createAndLoadRewardedAd();
        });

        mRewardedAdButton.setOnClickListener(v -> {

            Log.d(TAG, "rewardedAd.onClick()");

            if (mRewardedAd != null) {

                mRewardedAd.show(MainActivity.this, reward -> {
                    Log.d(TAG, "onUserEarnedReward of Type: " + reward.getType() + " and amount:" + reward.getAmount());

                    Toast.makeText(getApplicationContext(), "onUserEarnedReward of Type: " + reward.getType() + " and amount:" + reward.getAmount(), Toast.LENGTH_SHORT).show();
                });

            } else {
                Log.d(TAG, "The rewarded ad wasn't loaded yet.");
            }
        });
    }

    public void createAndLoadRewardedAd() {
        Log.d(TAG, "createAndLoadRewardedAd()");

        mRewardedAdButton.setVisibility(View.GONE);

        Bundle pollfishBundle = new PollfishExtrasBundleBuilder()
                .setAPIKey(POLLFISH_API_KEY)
                .setReleaseMode(false)
                .setRequestUUID("MY_ID")
                .build();

        AdRequest request = new AdRequest.Builder()
                .addNetworkExtrasBundle(PollfishAdMobAdapter.class, pollfishBundle)
                .build();

        RewardedAd.load(this, AD_MOB_KEY, request, new RewardedAdLoadCallback() {

            @Override
            public void onAdLoaded(@NonNull RewardedAd rewardedAd) {
                mRewardedAd = rewardedAd;

                mRewardedAdButton.setVisibility(View.VISIBLE);

                Toast.makeText(MainActivity.this, "onRewardedAdLoaded", Toast.LENGTH_SHORT).show();

                mRewardedAd.setFullScreenContentCallback(new FullScreenContentCallback() {

                    @Override
                    public void onAdFailedToShowFullScreenContent(@NonNull AdError adError) {
                        Log.d(TAG, "onRewardedAdFailedToShow");

                        Toast.makeText(getApplicationContext(), "onRewardedAdFailedToShow: " + adError.getMessage(), Toast.LENGTH_SHORT).show();
                    }

                    @Override
                    public void onAdShowedFullScreenContent() {
                        Log.d(TAG, "onRewardedAdOpened");

                        Toast.makeText(getApplicationContext(), "onRewardedAdOpened", Toast.LENGTH_SHORT).show();
                    }

                    @Override
                    public void onAdDismissedFullScreenContent() {
                        Log.d(TAG, "onRewardedAdClosed");

                        Toast.makeText(getApplicationContext(), "onRewardedAdClosed", Toast.LENGTH_SHORT).show();

                        createAndLoadRewardedAd();
                    }

                });

            }

            @Override
            public void onAdFailedToLoad(@NonNull LoadAdError loadAdError) {
                Toast.makeText(getApplicationContext(), "onRewardedAdFailedToLoad errorCode:" + loadAdError.getMessage(), Toast.LENGTH_SHORT).show();

                mRewardedAdButton.setVisibility(View.GONE);

                mRewardedAd = null;
            }

        });
    }

}

File: example-kotlin/src/main/java/com/pollfish/mediation/admob/example/MainActivity.kt
package com.pollfish.mediation.admob.example

import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.gms.ads.*
import com.google.android.gms.ads.rewarded.RewardItem
import com.google.android.gms.ads.rewarded.RewardedAd
import com.google.android.gms.ads.rewarded.RewardedAdLoadCallback
import com.pollfish.mediation.PollfishAdMobAdapter
import com.pollfish.mediation.PollfishExtrasBundleBuilder
import com.pollfish.mediation.admob.example.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    companion object {
        const val TAG = "MainActivity"
        const val AD_MOB_AD_UNIT_KEY = "ADMOB_AD_UNIT_KEY"
        const val POLLFISH_API_KEY = "POLLFISH_API_KEY"
    }

    private var mRewardedAd: RewardedAd? = null
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        MobileAds.initialize(this) {
            Log.d(TAG, "onInitializationComplete()")
            createAndLoadRewardedAd()
        }

        binding.rewardedAdBtn.setOnClickListener {
            Log.d(TAG, "rewardedAd.onClick()")

            if (mRewardedAd != null) {
                mRewardedAd!!.show(this@MainActivity) { reward: RewardItem ->
                    Log.d(
                        TAG,
                        "onUserEarnedReward of Type: ${reward.type} and amount: ${reward.amount}"
                    )
                    Toast.makeText(
                        applicationContext,
                        "onUserEarnedReward of Type: ${reward.type} and amount: ${reward.amount}",
                        Toast.LENGTH_SHORT
                    ).show()
                }
            } else {
                Log.d(TAG, "The rewarded ad wasn't loaded yet.")
            }
        }
    }

    private fun createAndLoadRewardedAd() {
        binding.rewardedAdBtn.isEnabled = false

        val bundle = PollfishExtrasBundleBuilder()
            .setAPIKey(POLLFISH_API_KEY)
            .setReleaseMode(false)
            .setRequestUUID("MY_UUID")
            .build()

        val request = AdRequest.Builder()
            .addNetworkExtrasBundle(PollfishAdMobAdapter::class.java, bundle)
            .build()

        RewardedAd.load(this, AD_MOB_AD_UNIT_KEY, request, object : RewardedAdLoadCallback() {
            override fun onAdLoaded(rewardedAd: RewardedAd) {
                mRewardedAd = rewardedAd
                binding.rewardedAdBtn.isEnabled = true
                Toast.makeText(this@MainActivity, "onRewardedAdLoaded", Toast.LENGTH_SHORT).show()
                rewardedAd.fullScreenContentCallback = object : FullScreenContentCallback() {
                    override fun onAdFailedToShowFullScreenContent(adError: AdError) {
                        Log.d(TAG, "onRewardedAdFailedToShow")
                        Toast.makeText(
                            applicationContext,
                            "onRewardedAdFailedToShow: ${adError.message}",
                            Toast.LENGTH_SHORT
                        ).show()
                    }

                    override fun onAdShowedFullScreenContent() {
                        Log.d(TAG, "onRewardedAdOpened")
                        Toast.makeText(applicationContext, "onRewardedAdOpened", Toast.LENGTH_SHORT)
                            .show()
                    }

                    override fun onAdDismissedFullScreenContent() {
                        Log.d(TAG, "onRewardedAdClosed")
                        Toast.makeText(applicationContext, "onRewardedAdClosed", Toast.LENGTH_SHORT)
                            .show()
                        createAndLoadRewardedAd()
                    }
                }
            }

            override fun onAdFailedToLoad(loadAdError: LoadAdError) {
                Toast.makeText(
                    applicationContext,
                    "onRewardedAdFailedToLoad errorCode: ${loadAdError.message}",
                    Toast.LENGTH_SHORT
                ).show()
                binding.rewardedAdBtn.isEnabled = false
                mRewardedAd = null
            }
        })
    }
}